sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_challenge_auth_policy.py

```diff
@@ -23,27 +26,48 @@ class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
                 # no_body was created with request's headers -> if request has a body, no_body's content-length is wrong
                 no_body.headers["Content-Length"] = "0"
 
+            _LOGGER.info("Provoking challenge with unauthorized bodiless request to {} ".format(
+                request.http_request.url
+            ))
             challenger = await self.next.send(PipelineRequest(http_request=no_body, context=request.context))
             try:
                 challenge = self._update_challenge(request, challenger)
             except ValueError:
                 # didn't receive the expected challenge -> nothing more this policy can do
+                _LOGGER.debug("Did not receive the expected challenge.")
                 return challenger
 
         await self._handle_challenge(request, challenge)
+
+        self._log_request(request)
         response = await self.next.send(request)
 
+        _LOGGER.info("Received response with status code {} {} from request to {}".format(
+            response.http_response.status_code,
+            response.http_response.reason,
+            response.http_request.url
+        ))
+
         if response.http_response.status_code == 401:
             # cached challenge could be outdated; maybe this response has a new one?
             try:
                 challenge = self._update_challenge(request, response)
             except ValueError:
                 # 401 with no legible challenge -> nothing more this policy can do
+                _LOGGER.debug("Received a 401 with no legible challenge.")
```

Comment:
This is the same as the above `ValueError` case, an unexpected response this policy can't handle and will return to the client, which will likely raise.

Reply:
Removed the ValueError debugging in both instances